### PR TITLE
Update variables in `inputref.py`

### LINKED ISSUE #77: Update `MISSING_MONTHLY_VARS`
I needed to download monthly `10m_u_component_of_wind` and era5cli wouldn't let me. Upon removing it from the list of missing variables in `inputref.py`, it worked without a problem.

That list should probably be updated; moreover it would be good to add a comment where it is coming from and how it can be kept up-to-date in the future.

### PR DESCRIPTION
**Pull request by bvreede**
The current variables in `inputref.py` do not reflect the current state of the API at CDS. The file has been amended to include more variables, both hourly and monthly, that are available for the ERA5 datasets.

Three variables were in our reference, but not part of the selection at CDS:
-    `altimeter_corrected_wave_height`
-    `altimeter_range_relative_correction`
-    `altimeter_wave_height`

I have left them in (running a quick check with one of these showed that they did in fact result in the download of data), but propose that we take another look at the current set of variables, perhaps consider how we can automate this workflow so we have an up-to-date set of variables in the tool (see #83), and/or consider how much we want era5cli to 'protect' the user from framing an impossible request.

I have also generated a list of variables that is complementary to `MISSING_MONTHLY_VARS`, as there were several variables in the monthly datasets that were not found in the hourly datasets. However, while the `MISSING_MONTHLY_VARS` are used in checks inside the tool (to make sure that a user does not create an 'impossible' request with these variables), the `MISSING_HOURLY_VARS` are not. The existence of this list is, at the moment, purely for documentation.

Closes #77.

### CODE REVIEW

**Review by stefsmeets — CHANGES_REQUESTED** [review]
So, if I understand correctly, there is a master list of variables: `SLVARS`, and then an additional list of variables which are missing from the master list, so that `HOURLY_VARS == SLVARS - MISSING_HOURLY_VARS`?

Then, should `fetch.py` also be updated somewhere here to make use of this new `MISSING_HOURLY_VARS` dict?

**Comment by bvreede** [comment]
Thanks @stefsmeets, well spotted!

I chose not to do this, as the purpose of `MISSING_HOURLY_VARS` at the moment is purely for documentation. Implementing this list will have effects on the cli calls done that include these variables, and it might even break some workflows; I have not checked whether these variables actually do return data, but they could; implementing this list in the same way as done with `MISSING_MONTHLY_VARS` will stop the request from being formed altogether. There must be a reason why they're gone from CDS, but e.g. I did find other variables that were no longer on the list on the website, but when used in an era5cli call still return data.

As for the variable list itself: there is a bigger problem there, too, in that this is a (currently purely manual) update that should be done frequently with CDS updates. In other words: this list of variables needs to be refactored into an automated workflow (see also #83), so I did not want to spend too much energy into implementing any downstream refactoring that may become obsolete soon.

The main point for this PR is to make sure there are no bugs in the upcoming release, so I tried to match the variable lists with info on CDS, and added the variable for documentation purposes. I've now tried to make this clearer in the code itself.

File: era5cli/inputref.py [hunk 1]
```diff
@@ -17,8 +17,32 @@
...
+# Pressure levels variables via:
+# reanalysis-era5-pressure-levels
+# reanalysis-era5-pressure-levels-monthly-means
+# reanalysis-era5-pressure-levels-preliminary-back-extension
+# reanalysis-era5-pressure-levels-monthly-means-preliminary-back-extension
+
+# ERA5-land variables via:
+# reanalysis-era5-land
+# reanalysis-era5-land-monthly-means
+
```
[the remark this answers is not in the document]
Comment (bvreede) [reply era5cli/inputref.py]: Yep thanks! Done!

**Review by stefsmeets — APPROVED** [review]
Alright, just making sure it was not an oversight! 🚀

**Pull request merged**